cmd/jimmsrv/service/service.go

```diff
@@ -499,21 +562,17 @@ func (s *Service) setupDischarger(p Params) (*discharger.MacaroonDischarger, err
 		MacaroonExpiryDuration: p.MacaroonExpiryDuration,
 		ControllerUUID:         p.ControllerUUID,
 	}
-	MacaroonDischarger, err := discharger.NewMacaroonDischarger(cfg, &s.jimm.Database, s.jimm.OpenFGAClient)
+	MacaroonDischarger, err := discharger.NewMacaroonDischarger(cfg, s.jimm.Database, s.jimm.OpenFGAClient)
 	if err != nil {
 		return nil, errors.E(err)
 	}
 	return MacaroonDischarger, nil
 }
 
-func (s *Service) setupSessionStore(ctx context.Context, sessionSecret []byte) (*pgstore.PGStore, error) {
+func (s *Service) setupSessionStore(ctx context.Context, sessionSecret []byte, db *db.Database) (*pgstore.PGStore, error) {
 	const op = errors.Op("setupSessionStore")
 
-	if s.jimm.CredentialStore == nil {
-		return nil, errors.E(op, "credential store is not configured")
-	}
-
-	sqlDb, err := s.jimm.Database.DB.DB()
+	sqlDb, err := db.DB.DB()
```

Comment:
Can we improve this at the moment?

Reply:
improve how?

Reply:
For a separate PR, and not for you per se, but I was just remarking that the naming isn't great. database.database.database() is crazy.